packages/common/src/search/_internal/hubSearchEventAttendees.ts

```diff
@@ -0,0 +1,144 @@
+import { getRegistrations } from "../../events/api/registrations";
+import { eventAttendeeToSearchResult } from "../../events/api/utils/search";
+import { IPredicate, IQuery } from "../types/IHubCatalog";
+import { IHubSearchOptions } from "../types/IHubSearchOptions";
+import { IHubSearchResponse } from "../types/IHubSearchResponse";
+import { IHubSearchResult } from "../types/IHubSearchResult";
+import {
+  GetRegistrationsParams,
+  IGetRegistrationsParams,
+  IPagedRegistrationResponse,
+} from "../../events/api/types";
+
+export const processIGetRegistrationsParams = (
+  options: IHubSearchOptions,
+  query: IQuery
+): IGetRegistrationsParams => {
+  // valid keys to translate from IQuery predicates
+  const validPredicateKeys: Array<keyof IPredicate> = [
+    // "term",
+    "userId",
+    "role",
+    "status",
+    "type",
+    "updatedRange",
+  ];
+  // valid keys to translate from IHubSearchOptions
+  const validPaginationKeys: Array<keyof IHubSearchOptions> = [
+    "num",
+    "start",
+    "sortField",
+    "sortOrder",
+  ];
+  // mappings for keys into GetRegistrationOptions equivalents
+  const keyMappings: Record<
+    keyof IHubSearchOptions | keyof IPredicate,
+    keyof GetRegistrationsParams
+  > = {
+    sortField: "sortBy",
+    attendanceType: "type",
+    // term: "name"
+  };
+  // translate IQuery filters into GetRegistrationParams
+  const filterProps: Record<string, string> = {};
```

Comment:
This method is a little difficult to read/understand. The sort-related properties are on IHubSearchOptions, not the IQuery, so this method needs to receive that object. Ideally, we'd re-use the same keys for the sort-related props that are used when searching for other types of entities via the hubSearch method. I'd consider simplifying this by:
1. abstract out the logic for processing filters from the IQuery into a separate `processAttendeeFilters` method
2. abstract out the logic for processing sort params from the IHubSearchOptions into a separate `processAttendeeOptions` method
3. Explicitly check for and transform specific params rather than looping over them all and conditionally doing different things based on their key within the loop

Borrowing from what I did for searching for events, this could look something like the following (disclaimer, I did not test this, but typescript seems to think this works).

```ts
// packages/common/src/search/_internal/hubEventsHelpers/processAttendeeOptions.ts
// test this in isolation

export function processAttendeeOptions(
  options: IHubSearchOptions
): Partial<GetRegistrationsParams> {
  const processedOptions: Partial<GetRegistrationsParams> = {};
  if (options.num > 0) {
    processedOptions.num = options.num.toString();
  }
  if (options.start > 1) {
    processedOptions.start = options.start.toString();
  }
  if (options.sortField === "modified") {
    processedOptions.sortBy = RegistrationSort.updatedAt;
  } else if (options.sortField === "created") {
    processedOptions.sortBy = RegistrationSort.createdAt;
  } else if (options.sortField === "username") {
    processedOptions.sortBy = RegistrationSort.username;
  } else if (options.sortField === "firstName") {
    processedOptions.sortBy = RegistrationSort.firstName;
  } else if (options.sortField === "lastName") {
    processedOptions.sortBy = RegistrationSort.lastName;
  }
  processedOptions.sortOrder =
    options.sortOrder === "desc" ? SortOrder.desc : SortOrder.asc;
  return processedOptions;
}
```

```ts
// packages/common/src/search/_internal/hubEventsHelpers/processAttendeeFilters.ts
// test this in isolation

export function processAttendeeFilters(query: IQuery): Partial<GetRegistrationsParams> {
  const processedFilters: Partial<GetRegistrationsParams> = {
    eventId: query.properties.eventId,
  };
  const useElseJoin = (value: string, defaults: string[]): string => value?.length
    ? value
    : defaults
      .map(val => val.toLowerCase())
      .join(',');

  const term = getPredicateValuesByKey(query.filters, "term");
  if (term.length) {
    // TODO: remove ts-ignore once GetEventsParams supports filtering by username, firstName, lastName https://devtopia.esri.com/dc/hub/issues/10153
    // @ts-ignore
    processedFilters.name = term[0];
  }

  processedFilters.type = useElseJoin(
    getOptionalPredicateStringsByKey(
      query.filters,
      "attendanceType"
    ),
    [EventAttendanceType.VIRTUAL, EventAttendanceType.IN_PERSON]
  );

  processedFilters.role = useElseJoin(
    getOptionalPredicateStringsByKey(query.filters, "role"),
    [RegistrationRole.OWNER, RegistrationRole.ORGANIZER, RegistrationRole.ATTENDEE]
  );

  processedFilters.status = useElseJoin(
    getOptionalPredicateStringsByKey(query.filters, "status"),
    [RegistrationStatus.PENDING, RegistrationStatus.ACCEPTED, RegistrationStatus.DECLINED, RegistrationStatus.BLOCKED]
  );
  
  const updatedDateRange = getPredicateValuesByKey(query.filters, "updatedDateRange");
  if (updatedDateRange.length) {
    processedFilters.updatedAtBefore = new Date(
      updatedDateRange[0].to
    ).toISOString();
    processedFilters.updatedAtAfter = new Date(
      updatedDateRange[0].from
    ).toISOString();
  }

  return processedFilters;
}
```

```ts
// packages/common/src/search/_internal/hubSearchEventAttendees.ts
// test this in isolation, since other methods are pulled out into separate modules we can spy/stub them to return anything to simplify testing of this method, lmk if you have questions

export async function hubSearchEventAttendees(
  query: IQuery,
  options: IHubSearchOptions
): Promise<IHubSearchResponse<IHubSearchResult>> {
  const processedFilters = processAttendeeFilters(query);
  const processedOptions = processAttendeeOptions(options);
  const data: GetRegistrationsParams = {
    ...processedFilters,
    ...processedOptions,
  };
  const { items, nextStart, total } = await getRegistrations({
    ...options.requestOptions,
    data,
  });
  const results = await Promise.all(
    items.map((eventAttendee) =>
      eventAttendeeToSearchResult(eventAttendee, options)
    )
  );
  const hasNext = nextStart > -1;
  return {
    total,
    results,
    hasNext,
    next: () => {
      if (!hasNext) {
        throw new Error("No more hub events for the given query and options");
      }
      return hubSearchEventAttendees(query, {
        ...options,
        start: nextStart,
      });
    },
  };
}
```